File: python/task8.py
class Solution:
    def canJump(self, nums: list[int]) -> bool:
        aim = len(nums)-1
        for i in range(len(nums)-2, -1, -1):
            if i + nums[i] >= aim:
                aim = i

        if aim == 0:
            return True
        return False

File: c++/algorithms/task8.cpp
#include <vector>

class Solution {
public:
    bool canJump(std::vector<int>& nums) {
        int aim = nums.size()-1;
        for (int i = nums.size()-2; i >= 0; i--){
            if (i + nums[i] >= aim){
                aim = i;
            }
        }
        if (aim == 0){
            return true;
        }
        return false;
    }
};